src/components/App.js

```diff
@@ -173,6 +169,38 @@ class App extends Component<{}, State> {
     setNotes(copy)
   }
 
+  removeFolder = (folder: Folder) => {
+    const archiveCopy = this.state.archive.slice()
+    const notesCopy = this.state.notes.slice()
+    const archiveTemp = []
+    this.state.notes.forEach(note => {
+      if (note.folder.name === folder) {
+        const index = _.findIndex(notesCopy, {
+          key: note.key,
+        })
+        notesCopy.splice(index, 1)
+        const noteCopy = note
+        noteCopy.folder.name = 'archive'
+        noteCopy.folder.type = 'archive'
```

Comment:
I think this pair of assignments also happens in `updateArchive`. One possible way to tighten it up: create a constant folder, e.g. `const archiveFolder = { name: "archive", type: "archive" }` and then directly assign the folder, e.g. `noteCopy.folder = archiveFolder`.